theme/snippets/add-to-cart-container.liquid

```diff
@@ -0,0 +1,47 @@
+<!-- Assign label and disabled state to add to cart button -->
+{% capture button_label %}{{"snippets.add_to_cart_container.add_to_cart_btn.text" | t }}{% endcapture %}
+{% assign button_disabled = false %}
+{% if product.available %}
+  {% if product_in_cart %}
+    {% capture button_label %}{{"snippets.add_to_cart_container.added_to_cart_btn.text" | t }}{% endcapture %}
+    {% assign button_disabled = true %}
+  {% endif %}
+{% else %}
+  {% case product_type %}
+    {% when "Course" %}
+    {% when "Event" %}
+      {% capture button_label %}{{"snippets.add_to_cart_container.registration_closed_btn.text" | t }}{% endcapture %}
+      {% assign button_disabled = true %}
+    {% else %}
+      {% capture button_label %}{{"snippets.add_to_cart_container.sold_out_btn.text" | t }}{% endcapture %}
+      {% assign button_disabled = true %}
+  {% endcase %}
+{% endif %}
+
+<div class="AddToCartContainer">
+  <form data-add-to-cart-form data-product-type={{product_type}} action="/cart/add" method="post" enctype="multipart/form-data" id="AddToCartForm">
+    <div class="AddToCartContainer__quantity-select-wrapper overflow-hidden relative mt_625{% if product.variants.size <= 1 %} mb1_75 {% endif %}">
+      <select class="sans-xs uppercase radius-xs border-black w100 pl1_125 py_5" name="quantity" id="Quantity" data-quantity-selector>
+        <option value="" selected="selected">{{"snippets.add_to_cart_container.quantity_spinner.label" | t }}</option>
+        {% for i in (1...10) limit: product.selected_or_first_available_variant.inventory_quantity %}
+          <option value="{{ i }}">{{ i }}</option>
+        {% endfor %}       
+      </select>
+    </div>
+    <div class="AddToCartContainer__variant-select-wrapper overflow-hidden relative mt_625 mb1_75 {% if product.variants.size <= 1 %} none {% endif %}">
+      <select class="sans-xs uppercase radius-xs border-black w100 pl1_125 py_5 {% if product.variants.size <= 1 %} none {% endif %}" name="variant" data-variant-selector>
+        {% if product.variants.size > 1 %}
+          <option value="" selected="selected">{{"snippets.add_to_cart_container.variant_select.label" | t }}</option>
+        {% endif %}
+        {% for variant in product.variants %}
+          <option value="{{ variant.id }}">
+            {{ variant.title }}
+          </option>
+        {% endfor %}
+      </select>
+    </div>
+    <button data-variant-id={{ product.first_available_variant.id }} data-add-to-cart-button {% if button_disabled %} disabled {% endif %} class="w100 color-white uppercase sans-light-sm radius-xs bg-color-black py_75" type="submit" name="add" id="AddToCart">
```

Comment:
This works but there are some issues that could use your 🧠 @sepowitz!

- The `add to cart button `only changes to "added to cart" if the product added is the `first_available_variant` (instead of the current product selected from the variant dropdown). This happens because in `cart.ts` we check if `(addToCartVariantId === variantId)`, then `add to cart button` shows "added to cart". Should I be trying to get the current product selected from the variant dropdown and pass that as `data-variant-id`?

- If there are two variants, small and medium, and you add a small item to your cart, the add to cart button changes to "added to cart" and is disabled, preventing you from adding other variants. Is there a better option to send feedback to the user that their product has been added, instead of disabling the button, now that we have products with multiple variants? Or could we do something like have a timer and enable the button again after a few seconds?